src/syscheckd/db/src/fimDB.cpp

```diff
@@ -166,3 +152,35 @@ void FIMDB::executeQuery(const nlohmann::json& item, ResultCallbackData callback
 {
     m_dbsyncHandler->selectRows(item, callbackData);
 }
+
+void FIMDB::fimRunIntegrity()
+{
+    std::unique_lock<std::mutex> lock{m_fimSyncMutex};
+
+    registerRSync();
+    loopRSync(lock);
+}
+
+void FIMDB::fimSyncPushMsg(const std::string& data)
+{
+    std::unique_lock<std::mutex> lock{m_fimSyncMutex};
+
+    if (!m_stopping)
+    {
+        auto rawData{data};
+        const auto buff{reinterpret_cast<const uint8_t*>(rawData.c_str())};
```

Comment:
why do you use `uint8_t*` here?

Reply:
It is needed to cast the parameters uint8_t, to pass into the function pushMessage.